src/pages/auth/Signup.jsx

```diff
@@ -2,36 +2,48 @@ import "./auth.css";
 
 import React,{useState} from "react";
 import { Link } from "react-router-dom";
+import { signinHandler } from "../../redux/reducers/authSlice";
+import { useDispatch,useSelector } from "react-redux";
 
 export const Signup = () => {
+  const dispatch = useDispatch();
     const [isEyeOpen, setOpen] = useState(false);
+    const [details,setDetails] = useState({email:"",password:""})
+    const [confirmPass,setConfirmPass] = useState("")
+    const {authenticating} = useSelector(store => store.auth)
+    const changeHandler = (e)=>{
+      setDetails(prev => ({...prev,[e.target.name]:e.target.value}))
+    }
+    const submitHandler = (e) =>{
+      e.preventDefault();
+      if(details.email === "" || details.password === "" || details.password !== confirmPass){
+        return;
+      }
+      dispatch(signinHandler(details))
+      setDetails(prev => ({...prev,email:"",password:""}))
+      setConfirmPass("")
+    }
   return (
     <div className="auth-wrapper">
       <h3 className="head">Register Now : </h3>
       <div className="auth-card">
-        <form action="signup.html">
-          <input
-            type="text"
-            name="name"
-            id="name"
-            className="input-field"
-            placeholder="Enter your full name"
-            required
-          />
+     
           <input
             type="email"
             name="email"
-            id="email"
+            value={details.email}
             className="input-field"
             placeholder="Enter your email here"
+            onChange={(e)=>changeHandler(e)}
             required
           />
           <input
             type={isEyeOpen ? "text" : "password"}
             name="password"
-            id="sign-pass"
+            value={details.password}
             className="input-field"
             placeholder="Enter your password here"
+            onChange={(e)=>changeHandler(e)}
```

Comment:
you can simplify it with `onChange={changeHandler}`

Reply:
I need event to be passed in function as well

Reply:
you can still receive the event in the arguments. I have done it this way.